app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+
+
+
+/**
+ * Class that contains the methods to read the allergens from a file, create an allergen list,
+ * find the allergens in the recognized ingredients from the OCR.
+ */
+class AllergensManager {
+
+    private final int COMMON_NAME = 0;
+    private final int INCI_NAMES = 1;
+
+    private ArrayList<Allergen> allergensList = new ArrayList<>();
+
+    private Context context;
+    private SharedPreferences sp = null;
+
+    /**
+     * constructor, accesses the sharedPreferences file and reads the allergens database from res
+     * @param cntxt context of the app/calling activity
+     * @author Cervo Nicolò
+     */
+    AllergensManager(Context cntxt) {
+
+        context = cntxt;
+        String selectedAllergensSP = cntxt.getString(R.string.selected_allergens_sharedPref);
+        sp = context.getSharedPreferences(selectedAllergensSP, Context.MODE_PRIVATE);
+        readAllergensdb();
+
+    }
+
+    /**
+     * Reads the allergendb.csv file from resources and stores the content in allergensList
+     * @author Cervo Nicolò
+     */
+    private  void readAllergensdb() {
+
+        // Parse through the allergendb.csv with CSVReader
+        InputStream inputStream = context.getResources().openRawResource(R.raw.allergendb);
+        InputStreamReader allergenDbReader = new InputStreamReader(inputStream);
+        CSVReader csvReader = new CSVReader(allergenDbReader);
+
```

Comment:
May it be a good practice to use the close() method to the csvReader when it will be no longer used?

Reply:
You are right.